metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1103,15 +1103,25 @@ def _build_input_measure_spec_for_base_metric(
                 offset_to_grain=child_metric_offset_to_grain,
             )
 
-        # Even if the measure is configured to join to time spine, if there's no metric_time in the query,
-        # there's no need to join to the time spine since all metric_time will be aggregated.
+        # Even if the measure is configured to join to time spine, if there's no agg_time_dimension in the query,
+        # there's no need to join to the time spine since all time will be aggregated.
         after_aggregation_time_spine_join_description = None
-        if input_measure.join_to_timespine and query_contains_metric_time:
-            after_aggregation_time_spine_join_description = JoinToTimeSpineDescription(
-                join_type=SqlJoinType.LEFT_OUTER,
-                offset_window=None,
-                offset_to_grain=None,
-            )
+        if input_measure.join_to_timespine:
+            query_contains_agg_time_dimension = queried_linkable_specs.contains_metric_time
+            if not query_contains_agg_time_dimension:
+                valid_agg_time_dimensions = self._semantic_model_lookup.get_agg_time_dimension_specs_for_measure(
+                    measure_spec.reference
+                )
+                query_contains_agg_time_dimension = bool(
+                    set(queried_linkable_specs.time_dimension_specs).intersection(set(valid_agg_time_dimensions))
+                )
+
+            if query_contains_agg_time_dimension:
```

Comment:
Yeah if we have some kind of helper I think it'll clean up a lot of these conditionals as well, because it'll hide the wonky logic inside of the helper and this can become something like:

`if query_contains_agg_time_dimension(queried_linkable_specs, measure_spec):`

Reply:
Updated!